cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SupportBundleHttpHandler.java

```diff
@@ -0,0 +1,568 @@
+          metrics.add(timeSeries.getMetricName(), new JsonArray());
+        }
+        for (MetricQueryResult.TimeValue timeValue : timeSeries.getData()) {
+          JsonObject time = new JsonObject();
+          time.addProperty("time", timeValue.getTime());
+          time.addProperty("value", timeValue.getValue());
+          metrics.getAsJsonArray(timeSeries.getMetricName()).add(time);
+        }
+      }
+      return metrics;
+    } catch (Exception e) {
+      LOG.error("Can not add metrics: ", e);
+      return new JsonObject();
+    }
+  }
+
+  /** Generates pipeline log file and run info file */
+  private void generateLogFileAndRunInfo(
+    ProgramRunId latestProgramRunId,
+    RunRecordDetail latestRunRecordDetail,
+    String appPath,
+    JsonObject metrics)
+    throws Exception {
+
+    // Generates pipeline log file for user request
+    CompletableFuture<Void> futurePipelineLog =
+      CompletableFuture.runAsync(
+        () -> {
+          try {
+            LoggingContext loggingContext =
+              LoggingContextHelper.getLoggingContextWithRunId(latestProgramRunId, null);
+            generateLogFile(loggingContext, appPath, latestProgramRunId.getRun() + "-log.txt");
+          } catch (Exception e) {
+            LOG.error("Can not write pipeline log file ", e);
+          }
+        });
+    futurePipelineLog.get();
+    addToStatus("runtimelog", appPath);
+    // Generate runtime info details file
+    CompletableFuture<Void> futureRuntimeInfo =
+      CompletableFuture.runAsync(
+        () -> {
+          try (FileWriter file =
+                 new FileWriter(appPath + "/" + latestProgramRunId.getRun() + ".json")) {
+            JsonObject jsonObject = new JsonObject();
+            jsonObject.addProperty("status", latestRunRecordDetail.getStatus().toString());
+            jsonObject.addProperty("start", latestRunRecordDetail.getStartTs());
+            jsonObject.addProperty("end", latestRunRecordDetail.getStopTs());
+            jsonObject.addProperty(
+              "profileName", latestRunRecordDetail.getProfileId().getProfile());
+            jsonObject.addProperty(
+              "runtimeArgs", latestRunRecordDetail.getProperties().get("runtimeArgs"));
+            jsonObject.add("metrics", metrics);
+            file.write(GSON.toJson(jsonObject));
+            file.flush();
+          } catch (IOException e) {
+            LOG.error("Can not write runtime file ", e);
+          }
+        });
+    futureRuntimeInfo.get();
+    addToStatus("runtimeinfo", appPath);
+  }
+
+  /** Generates log file within certain path */
+  private void generateLogFile(LoggingContext loggingContext, String basePath, String filePath) {
+    try (FileWriter file = new FileWriter(basePath + "/" + filePath)) {
+      long currentTimeMillis = System.currentTimeMillis();
+      long fromMillis = currentTimeMillis - TimeUnit.DAYS.toMillis(1);
+      CloseableIterator<LogEvent> logIter =
```

Comment:
You have to close the `CloseableIterator` whenever you see one. Use the try (resource) construct to do so.